iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift

```diff
@@ -0,0 +1,282 @@
+    }
+
+    func setupBackground() {
+        UIGraphicsBeginImageContext(view.frame.size)
+        if traitCollection.userInterfaceStyle == .dark {
+            UIImage(named: "dayBackground")?.draw(in: view.bounds)
+        }
+        else {
+            UIImage(named: "nightBackground")?.draw(in: view.bounds)
+        }
+        let image: UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+        UIGraphicsEndImageContext()
+        view.backgroundColor = UIColor(patternImage: image)
+    }
+
+    func setupCityNameLabel() {
+        view.addSubview(cityNameLabel)
+        cityNameLabel.font = UIFont(name: "Roboto", size: 28)
+        cityNameLabel.textColor = .white
+
+        cityNameLabel.text = "Rīga"
+
+        cityNameLabel.snp.makeConstraints { make in
+            make.leading.equalTo(view.safeAreaLayoutGuide).offset(20)
+            make.top.equalTo(view).offset(120)
+        }
+    }
+
+    func setupTemperatureLabel() {
+        view.addSubview(temperatureLabel)
+        temperatureLabel.font = UIFont(name: "Roboto", size: 128)
+        temperatureLabel.textColor = .white
+
+        temperatureLabel.text = "2°"
+
+        temperatureLabel.snp.makeConstraints { make in
+            make.leading.equalTo(view.safeAreaLayoutGuide).offset(20)
+            make.top.equalTo(cityNameLabel.snp.bottom)
+        }
+    }
+
+    func setupSkyConditionLabel() {
+        view.addSubview(skyConditionLabel)
+        skyConditionLabel.font = UIFont(name: "BebasNeue", size: 24)
+        skyConditionLabel.transform = CGAffineTransform(rotationAngle: -CGFloat.pi / 2)
+        skyConditionLabel.textColor = .white
+
+        skyConditionLabel.text = "clear sky"
+
+        skyConditionLabel.snp.makeConstraints { make in
+            make.trailing.equalTo(view.safeAreaLayoutGuide).offset(15)
+            make.top.equalTo(view.snp.centerY).offset(60)
+        }
+    }
+
+    func setupDetailsStackView() {
+        view.addSubview(detailsStackView)
+        detailsStackView.backgroundColor = UIColor(red: 200.0/255.0, green: 255.0/255.0, blue: 255.0/255.0, alpha: 0.3)
+        detailsStackView.axis = .horizontal
+        detailsStackView.distribution = .equalCentering
+        detailsStackView.layoutMargins = UIEdgeInsets(top: 20, left: 30, bottom: 15, right: 30)
+        detailsStackView.isLayoutMarginsRelativeArrangement = true
+        detailsStackView.layer.cornerRadius = 20
+        detailsStackView.layer.borderWidth = 2
+        detailsStackView.layer.borderColor = UIColor.white.cgColor
+
+        detailsStackView.snp.makeConstraints { make in
+            make.left.equalTo(view.safeAreaLayoutGuide).offset(30)
+            make.right.equalTo(view.safeAreaLayoutGuide).inset(30)
+            make.height.equalTo(80)
+            make.bottom.equalTo(view.safeAreaLayoutGuide).inset(20)
+        }
+
+        setupDetailStackWindLabel()
+        setupDetailStackHumidityLabel()
+        setupDetailStackMaxTempLabel()
+    }
+
```

Comment:
try to create an collectionView with cell, at my logic, at future, we will have some information about previous time

Reply:
Plan for me for the main view(Weather Controller) is just showing current weather using current location and later when you press button it opens detailed TableView with custom cell where it's showing all info for next days. I thought since it's anyway list then it would be easier to work with tableview and not collection view.

Reply:
okey, use pls stackViews for settings ui at this vc

Reply:
I am using stackViews here. Vertical stacks inside horizontal stack